terraform/aws/main.tf

```diff
@@ -0,0 +1,58 @@
+provider "aws" {
+    region = "us-east-1"
+}
+
+resource "aws_instance" "approzium" {
+    ami = var.ami
+    instance_type = var.instance_type
+    user_data = templatefile(
+        "${path.module}/scripts/install.sh.tpl",
+        {
+        download-url  = var.download-url
+        config        = file("approzium.config.yml")
+        extra-install = var.extra-install
+        logs-file     = var.logs-file
+        }
+    )
+
+    tags = {
+        Name = var.instance_name
+    }
+
+    security_groups = [aws_security_group.approzium.name]
+    key_name = var.key-name
+}
+
+
+// Security group for Approzium allows SSH, HTTP, and GRPC access
+resource "aws_security_group" "approzium" {
+    name = "approzium"
+    description = "Approzium servers"
+}
+
+resource "aws_security_group_rule" "approzium-ssh" {
+    security_group_id = aws_security_group.approzium.id
+    type = "ingress"
+    from_port = 22
+    to_port = 22
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
```

Comment:
My gut sense is that by default we shouldn't allow folks to SSH into the instance hosting Approzium because then they can do lots of bad things like memory dumping to learn passwords. I should have linked this before, my apologies, there's a [Production Hardening Guide](https://learn.hashicorp.com/vault/operations/production-hardening) that probably gives a good idea of what we'd want for default behavior here. Totally forgot to mention this.

Reply:
No worries! We can do that. I was following the Vault terraform which allowed it, but I am in favor of sticking to the essentials and letting the user customize/add if they want.